src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,167 @@
+const REQUEST_TIMEOUT = 8000; // ms
+const request = require('request-promise-native');
+
+// STACKOVERFLOW BEAUTY
+function validURL(str) {
+	const pattern = new RegExp(
+		'^(https?:\\/\\/)?'
+		+ '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'
+		+ '((\\d{1,3}\\.){3}\\d{1,3}))'
+		+ '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'
+		+ '(\\?[;&a-z\\d%_.~+=-]*)?'
+		+ '(\\#[-a-z\\d_]*)?$',
+		'i',
+	);
+	return !!pattern.test(str);
+}
+
+class EduSharingConnector {
+	constructor() {
+		if (EduSharingConnector.instance) {
+			return EduSharingConnector.instance;
+		}
+		if (!validURL(this.url)) {
+			return 'Invalid ES_DOMAIN, check your .env';
+		}
+		this.authorization = null; // JSESSION COOKIE
+		this.accessToken = null; // ACCESSTOKEN
+		EduSharingConnector.instance = this;
+	}
+
+	static get headers() {
+		return {
+			Accept: 'application/json',
+			'Content-type': 'application/json',
+		};
+	}
+
+	static get authorization() {
+		const userName = process.env.ES_USER;
+		const pw = process.env.ES_PASSWORD;
+		const headers = Object.assign({}, EduSharingConnector.headers, {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use an object spread instead of `Object.assign` eg: `{ ...foo }`.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4791331)